engine/class_modules/sc_hunter.cpp

```diff
@@ -1,6031 +1,6034 @@
-    hunter_ranged_attack_t( n, p, p -> talents.volley -> effectN( 1 ).trigger() )
-  {
-    background = true;
-    aoe = -1;
-  }
-};
-
-// Auto Shot ================================================================
-
-struct auto_shot_t: public hunter_action_t < ranged_attack_t >
-{
-  volley_t* volley = nullptr;
-  bool first_shot = true;
-
-  auto_shot_t( hunter_t* p ):
-    hunter_action_t( "auto_shot", p, p -> find_spell( 75 ) )
-  {
-    background = true;
-    repeating = true;
-    interrupt_auto_attack = false;
-    special = false;
-    trigger_gcd = timespan_t::zero();
-
-    weapon = &( p -> main_hand_weapon );
-    base_execute_time = weapon -> swing_time;
-
-    if ( p -> talents.volley -> ok() )
-    {
-      volley = p -> get_background_action<volley_t>( "volley" );
-      add_child( volley );
-    }
-  }
-
-  void reset() override
-  {
-    hunter_action_t::reset();
-    first_shot = true;
-  }
-
-  timespan_t execute_time() const override
-  {
-    if ( first_shot )
-      return timespan_t::from_millis( 100 );
-    return hunter_action_t::execute_time();
-  }
-
-  void execute() override
-  {
-    if ( first_shot )
-      first_shot = false;
-
-    hunter_action_t::execute();
-  }
-
-  void impact( action_state_t* s ) override
-  {
-    hunter_action_t::impact( s );
-
-    if ( p() -> buffs.lock_and_load -> trigger() )
-      p() -> cooldowns.aimed_shot -> reset( true );
-
-    if ( s -> result == RESULT_CRIT && p() -> specialization() == HUNTER_BEAST_MASTERY )
-    {
-      double wild_call_chance = p() -> specs.wild_call -> proc_chance() +
-                                p() -> talents.one_with_the_pack -> effectN( 1 ).percent();
-
-      if ( rng().roll( wild_call_chance ) )
-      {
-        p() -> cooldowns.barbed_shot -> reset( true );
-        p() -> procs.wild_call -> occur();
-      }
-    }
-
-    if ( volley && rng().roll( p() -> talents.volley -> proc_chance() ) )
-    {
-      volley -> set_target( s -> target );
-      volley -> execute();
-    }
-  }
-};
-
-struct start_attack_t: public action_t
-{
-  start_attack_t( hunter_t* p, const std::string& options_str ):
-    action_t( ACTION_OTHER, "start_auto_shot", p )
-  {
-    parse_options( options_str );
-
-    p -> main_hand_attack = new auto_shot_t( p );
-    stats = p -> main_hand_attack -> stats;
-    ignore_false_positive = true;
-
-    trigger_gcd = timespan_t::zero();
-  }
-
-  void execute() override
-  {
-    player -> main_hand_attack -> schedule_execute();
-  }
-
-  bool ready() override
-  { return ! target -> is_sleeping() && player -> main_hand_attack -> execute_event == nullptr; } // not swinging
-};
-
-
-//==============================
-// Shared attacks
-//==============================
-
-// Barrage ==================================================================
-
-struct barrage_t: public hunter_spell_t
-{
-  struct barrage_damage_t: public hunter_ranged_attack_t
-  {
-    barrage_damage_t( const std::string& n, hunte
```

Comment:
You don't need `buff.aspect_of_the_eagle.up` checks as the actions themselves check that in `ready()`.